source/ldap.d

```diff
@@ -456,20 +456,23 @@ struct LDAPConnection
 			{
 				string attr_name = attr.to!string;
 
-				auto ppValue = ldap_get_values(_handle, entry, attr);
+				auto ppValue = ldap_get_values_len(_handle, entry, attr);
 				if (!ppValue)
 				{
 					result.attributes[attr_name] = [];
 				}
 				else
 				{
-					auto iValue = ldap_count_values(ppValue);
+					auto iValue = ldap_count_values_len(ppValue);
 					if (!iValue)
 						result.attributes[attr_name] = [];
 					else
-						result.attributes[attr_name] = ppValue[0 .. iValue].map!(a => a.to!(char[])
-								.idup).array;
-					ldap_value_free(ppValue);
+						result.attributes[attr_name] = 
+							ppValue[0 .. iValue].map!(a =>
+								a.bv_val[0..a.bv_len].to!(char[]).idup
```

Comment:
I think the `to!(char[])` is a no-op here, can you see if everything still works in your setup if you just remove it?

```suggestion
								a.bv_val[0..a.bv_len].idup
```